client/asset/btc/btc.go

```diff
@@ -2202,19 +2208,34 @@ func (btc *baseWallet) AuditContract(coinID, contract, txData dex.Bytes, rebroad
 		return nil, fmt.Errorf("error extracting swap addresses: %w", err)
 	}
 
-	// Perform basic validation using the txData.
-	// It may be worth separating data validation from coin
-	// retrieval at the asset.Wallet interface level.
-	tx, err := msgTxFromBytes(txData)
-	if err != nil {
-		return nil, fmt.Errorf("coin not found, and error encountered decoding tx data: %v", err)
-	}
-	if len(tx.TxOut) <= int(vout) {
-		return nil, fmt.Errorf("specified output %d not found in decoded tx %s", vout, txHash)
+	// If no tx data is provided, attempt to get the required data (the txOut)
+	// from the wallet. If this is a full node wallet, a simple gettxout RPC is
+	// sufficient with no pkScript or "since" time. If this is an SPV wallet,
+	// only a confirmed counterparty contract can be located, and only one
+	// within the last 48 hours. As such, this mode of operation is not intended
+	// for normal server-coordinated operation.
+	var tx *wire.MsgTx
+	var txOut *wire.TxOut
+	if len(txData) == 0 {
+		// Fall back to gettxout, but we won't have the tx to rebroadcast.
+		pkScript, _ := btc.scriptHashScript(contract) // pkScript and since time are unused if full node
+		txOut, _, err = btc.node.getTxOut(txHash, vout, pkScript, time.Now().Add(-48*time.Hour))
```

Comment:
These two comments are for https://github.com/decred/dcrdex/pull/1553, which this PR is stacked on. Will make the changes there.

Reply:
@buck54321 As a half measure, I made this an exported `var` so such oddball use cases (recovery tools and tests) can at least adjust the limit.  I think if we want this used more commonly, we could go the route of adding a `GetTxData` method as mentioned in the comments, and that could have this since time as an arg.